lsif/src/database.test.ts

```diff
@@ -0,0 +1,184 @@
+import { findRange, findResult, findMonikers, walkChain, asLocations, makeRemoteUri, comparePosition } from './database'
+import { Id, MonikerKind } from 'lsif-protocol'
+import { ResultSetData, RangeData, MonikerData } from './entities'
+import * as lsp from 'vscode-languageserver-protocol'
+
+describe('findRange', () => {
+    it('should find all ranges in list', () => {
+        // Generate starting characters for each range. Thse neds to be
+        // spread wide enough so that the ranges on each line don't touch.
+        const characters = Array.from(Array(2000).keys()).map(i => i * 5)
```

Comment:
Lodash makes this easier
```suggestion
        const characters = range(0, 10000, 5)
```

Reply:
Done